### PR TITLE
Audit/spot accounts

### PR DESCRIPTION
**Pull request by Thomas-Smets**

### CODE REVIEW

File: src/accounts/AccountSpot.sol [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+     * @notice Initiates the variables of the Account.
+     * @param owner_ The sender of the 'createAccount' on the Factory
+     * @param registry_ The 'beacon' contract with the external logic to price assets.
+     * @dev A proxy will be used to interact with the Account implementation.
+     * This function will only be called (once) in the same transaction as the proxy Account creation through the Factory.
+     * @dev initialize has implicitly a nonReentrant guard, since the "locked" variable has value zero until the end of the function.
+     * @dev The Registry is not used in spot accounts, but a valid registry must be set to be compatible with V1 Accounts.
+     */
+    function initialize(address owner_, address registry_, address) external onlyFactory {
+        if (registry_ == address(0)) revert AccountErrors.InvalidRegistry();
```
Comment (j-vp) [inline src/accounts/AccountSpot.sol:130]: Will there be a registry set in spot accounts?

Comment (Thomas-Smets) [reply src/accounts/AccountSpot.sol:130]: Yes, see @dev

File: src/accounts/helpers/SpotToMarginMigrator.sol [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+     * This will finalize the upgrade and transfer the Margin Account to the caller.
+     * @param account The Account to transfer back to its owner after the cool-down period.
+     */
+    function endUpgrade(address account) external {
+        if (accountToOwner[account] != msg.sender) revert NotOwner();
+        // Remove owner from Account in storage
+        accountToOwner[account] = address(0);
+        // Transfer Account back to owner
+        FACTORY.safeTransferFrom(address(this), msg.sender, account);
+    }
```
[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/accounts/helpers/SpotToMarginMigrator.sol:117]: We will probably not include this contract in the audit, but good point, will change

[the remark this answers is not in the document]
Comment (Thomas-Smets) [reply src/accounts/helpers/SpotToMarginMigrator.sol:117]: See https://github.com/arcadia-finance/accounts-v2/pull/244

**Review by acollette — APPROVED** [review]

**Pull request merged**